### ISSUE TITLE
How to make predictions using instance weights in xgboost?

### ISSUE THREAD

**Issue by hmzhan** [dmlc/xgboost#1523]
I created xgb.DMatrix like this, and it worked really well:
`dtrain <- xgb.DMatrix(data=train, label=label_train, weight=weights_train)`
`dtest <- xgb.DMatrix(data=test, label=label_test, weight=weights_test)`
`watchlist <- list(train=dtrain, test=dtest)`
`bst <- xgb.train(data=dtrain, watchlist=watchlist)`

when I used `predict` function in xgboost `predict(bst, dtest)`, the auc or other evaluation metrics were much smaller than those shown in training step. it seems like `predict` function does not use weights for the predictions.  Thanks!
